scripts/zones/Horlais_Peak/mobs/Evil_Oscar.lua

```diff
@@ -26,18 +26,29 @@ local ID = require("scripts/zones/Horlais_Peak/IDs")
 -----------------------------------
 local entity = {}
 
-local sendMessage = function(players)
+local sendFillingMessage = function(players)
     for _, member in pairs(players) do
         member:messageSpecial(ID.text.EVIL_OSCAR_BEGINS_FILLING)
     end
 end
 
+local sendOutOfRangeMessage = function(players, target)
+    for _, member in pairs(players) do
+        member:messageBasic(4, 0, 0, target)
+    end
+end
+
 local evilOscarFillsHisLungs
 evilOscarFillsHisLungs = function(mob)
     if not mob:isAlive() then
         return
     end
 
+    if not mob:isEngaged() then
+        mob:setLocalVar("EBB_BREATH_COUNTER", 0)
+        return
+    end
```

Comment:
Would this also work?
entity.onMobDisengage = function(mob)
    mob:resetLocalVars()
end

Reply:
> Would this also work? entity.onMobDisengage = function(mob) mob:resetLocalVars() end

The function calls itself so is only stopped when the function returns before getting to those self-calls thus the need for this check.